contracts/payment/SponsorPayment.sol

```diff
@@ -0,0 +1,519 @@
+    }
+
+    struct WithdrawalRequest {
+        uint256 amount;
+        uint256 lockTime;
+        bool exists;
+    }
+
+    /**
+     * @dev Main storage $tructure for the contract
+     */
+    struct SponsorPaymentStorage {
+        mapping(address => mapping(address => uint256)) balances; // sponsor => token => balance
+        mapping(address => mapping(address => WithdrawalRequest)) withdrawalRequests; // sponsor => token => request
+        mapping(bytes32 => bool) isWithdrawn;
+        uint8 ownerPercentage;
+        uint256 withdrawalDelay;
+    }
+
+    string public constant VERSION = "1.0.0";
+
+    bytes32 public constant ERC20_PAYMENT_CLAIM_DATA_TYPE_HASH =
+        keccak256(
+            // solhint-disable-next-line max-line-length
+            "ERC20SponsorPaymentInfo(address recipient,uint256 amount,address token,uint256 expiration,uint256 nonce,bytes metadata)"
+        );
+
+    bytes32 public constant PAYMENT_CLAIM_DATA_TYPE_HASH =
+        keccak256(
+            // solhint-disable-next-line max-line-length
+            "ERC20SponsorPaymentInfo(address recipient,uint256 amount,uint256 expiration,uint256 nonce,bytes metadata)"
+        );
+
+    //  keccak256(abi.encode(uint256(keccak256("iden3.storage.SponsorPayment")) - 1)) &
+    //             ~bytes32(uint256(0xff));
+    bytes32 private constant SPONSOR_PAYMENT_STORAGE_LOCATION =
+        0x98fc76e32452055302f77aa95cd08aa0cf22c02a3ebdaee3e1411f6c47c2ef00;
+
+    modifier validToken(address token) {
+        if (token != address(0)) {
+            if (!_isContract(token)) revert InvalidToken("Not a contract address");
+            SponsorPaymentStorage storage s = _getSponsorPaymentStorage();
+        }
+        _;
+    }
+
+    /**
+     * @dev Valid percent value modifier
+     */
+    modifier validPercentValue(uint256 percent) {
+        if (percent > 100) {
+            revert InvalidParameter("Invalid owner percentage");
+        }
+        _;
+    }
+
+    function initialize(
+        address owner,
+        uint8 ownerPercentage,
+        uint256 withdrawalDelay
+    ) external initializer validPercentValue(ownerPercentage) {
+        if (withdrawalDelay == 0) revert InvalidParameter("Invalid withdrawal delay");
+        SponsorPaymentStorage storage $ = _getSponsorPaymentStorage();
+        $.ownerPercentage = ownerPercentage;
+        $.withdrawalDelay = withdrawalDelay;
+        __ReentrancyGuard_init();
+        __EIP712_init("SponsorPayment", VERSION);
+        __Ownable_init(owner);
+    }
+
+    function _getSponsorPaymentStorage() private pure returns (SponsorPaymentStorage storage $) {
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            $.slot := SPONSOR_PAYMENT_STORAGE_LOCATION
+        }
+    }
+
+    /**
+     * @notice Checks if an address contains contract code
+     * @param addr Address to check
+     */
+    function _isContract(address addr) private view returns (bool) {
+        uint256 size;
```

Comment:
I guess it is the same check target.code.length  > 0  (you can test)

Reply:
replaced this check